sklearn/ensemble/bagging.py

```diff
@@ -276,40 +278,59 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("Out of bag estimation only available"
                              " if bootstrap=True")
 
-        # Free allocated memory, if any
-        self.estimators_ = None
+        if not self.warm_start:
+            # Free allocated memory, if any
+            self.estimators_ = []
 
-        # Parallel loop
-        n_jobs, n_estimators, starts = _partition_estimators(self.n_estimators,
-                                                             self.n_jobs)
-        seeds = random_state.randint(MAX_INT, size=self.n_estimators)
+        n_more_estimators = self.n_estimators - len(self.estimators_)
 
-        all_results = Parallel(n_jobs=n_jobs, verbose=self.verbose)(
-            delayed(_parallel_build_estimators)(
-                n_estimators[i],
-                self,
-                X,
-                y,
-                sample_weight,
-                seeds[starts[i]:starts[i + 1]],
-                verbose=self.verbose)
-            for i in range(n_jobs))
+        if n_more_estimators < 0:
```

Comment:
I'd just test for ``n_more_estimators <= 0`` and raise in either case. Nothing happens if ``n_more_estimators == 0``, right?